### PR TITLE
Add modals to bulk buttons

### PR DESCRIPTION
**Pull request by hmpf**
Dummy dialogs. The dialogs now pop up but nothing happens on submit.

### CODE REVIEW

File: src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
+        <li>{% include 'htmx/incidents/_incident_acknowledge_modal.html' with dialog_id="create-acknowledgment-dialog" button_class="btn-accent" button_title="Acknowledge" header="Submit acknowledgment" explanation="Write a message describing why these incidents were acknowledged" cancel_text="Cancel" submit_text="Submit" %}</li>
+        <li>{% include 'htmx/incidents/_incident_close_modal.html' with dialog_id="close_incident-dialog" button_class="btn-accent" button_title="Close" header="Manually close incidents" explanation="Write a message describing why these incidents were manually closed" cancel_text="Cancel" submit_text="Close now" %}</li>
+        <li>{% include 'htmx/incidents/_incident_reopen_modal.html' with dialog_id="reopen-incident-dialog" button_class="btn-accent" button_title="Reopen" header="Manually reopen incident" explanation="Write a message describing why these incident were manually reopened" cancel_text="Cancel" submit_text="Reopen now" %}</li>
```
Comment (johannaengland) [inline src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html:6]:
```suggestion
        <li>{% include 'htmx/incidents/_incident_reopen_modal.html' with dialog_id="reopen-incident-dialog" button_class="btn-accent" button_title="Reopen" header="Manually reopen incidents" explanation="Write a message describing why these incidents were manually reopened" cancel_text="Cancel" submit_text="Reopen now" %}</li>
```

**Review by johannaengland — COMMENTED** [review]
Not sure if this is a problem, but you said that nothing happens if you click on "Submit", but for me I get a white screen and in the console I get these error messages
```
2024-09-05 12:50:05,258 django.request WARNING  Method Not Allowed (POST): /incidents/
2024-09-05 12:50:05,258 django.server WARNING  "POST /incidents/?page=1&maxlevel=5 HTTP/1.1" 405 0
```

**Comment by johannaengland** [comment]
Also the way that the label looks is a bit strange, specifically the way the text wraps ("Change ticket", "Reason for closing" and "Reason for reopening"):
image
image
image

Acknowledgement looks good

File: src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html (hunk 1 above)
Comment (hmpf) [reply src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html:6]: Hoho, this is a problem, the templates are also used for the details page (where you only change one incident at a time).

Comment (hmpf) [reply src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html:6]: We won't need to have separate templates, phew.

**Comment by hmpf** [comment]
> Also the way that the label looks is a bit strange, specifically the way the text wraps ("Change ticket", "Reason for closing" and "Reason for reopening"): 
> /../
> 
> Acknowledgement looks good

Fix belongs in different issue. Either solved by having separate templates for long and short labels (with different styling), or changing the style for all modals so that the length of the label does not matter (my preference).

**Comment by hmpf** [comment]
> Not sure if this is a problem, but you said that nothing happens if you click on "Submit", but for me I get a white screen and in the console I get these error messages
> 
> ```
> 2024-09-05 12:50:05,258 django.request WARNING  Method Not Allowed (POST): /incidents/
> 2024-09-05 12:50:05,258 django.server WARNING  "POST /incidents/?page=1&maxlevel=5 HTTP/1.1" 405 0
> ```

I've allowed POST on the view and now this is avoided.

**Comment by hmpf** [comment]
I have no idea how to tell the bulk stuff which incidents have been selected, I'd rather do that in a separate PR.

This is not a "make the buttons do what they should"-PR but a "show the dialog on button press"-PR.

File: src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html [hunk 2]
```diff
@@ -1,7 +1,8 @@
 <ul class="menu menu-horizontal gap-2">
     {% block update_menu_content %}
-        <li><button class="btn btn-accent">Add Ticket</button></li>
-        <li><button class="btn btn-accent">Acknowledge</button></li>
-        <li><button class="btn btn-accent">Close/re-open</button></li>
+        <li>{% include 'htmx/incidents/_incident_ticket_edit_modal.html' with dialog_id="add-ticket-dialog" button_class="btn-accent" button_title="Change tickets" header="Change tickets" explanation="Write an URL of an existing ticket, or nothing to remove existing ticket urls" cancel_text="Cancel" submit_text="Submit" %}</li>
```
Comment (elfjes) [inline src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html:3]: `button_class` does not seem implemented in `_base_form_modal.html`, `btn-accent` does not seem to be applied

**Review by elfjes — CHANGES_REQUESTED** [review]
clicking while the modal is open seems to flash something. clicking and holding make this more apparent, I don't think it should do that:

image

File: src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html (hunk 2 above)
Comment (hmpf) [reply src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html:3]: What do you mean by `button_class`?

**Comment by hmpf** [comment]
> > I have no idea how to tell the bulk stuff which incidents have been selected, I'd rather do that in a separate PR.
> 
> I can push a commit for the first RFC then, or do you prefer a coding session together?

Push is fine. Should I move the branch to this repo?

File: src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html (hunk 2 above)
Comment (elfjes) [reply src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html:3]: `with dialog_id="add-ticket-dialog" button_class="btn-accent"`

Comment (hmpf) [reply src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html:3]: I could make the class (currently: "btn-primary") on the visible buttons overridable. The base modal is the same as is used in the details modals.

**Comment by hmpf** [comment]
I prefer separate PR's because quicker to review, shorter lived branches and doing only one thing.

File: src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html (hunk 2 above)
Comment (elfjes) [reply src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html:3]: Ok, maybe it's not clear what I mean. In the current main branch, the bulk buttons have `class="btn btn-accent"`. In this PR the bulk buttons only have `class="btn"` I thought that by passing a `button_class="btn-accent"` to the `include 'htmx/incidents/_incident_ticket_edit_modal.html` statement the idea was to place `btn-accent` on the bulk button. However, for this to work, inside  `_base_form_modal.html` the `button_class` context variable needs to be read, something like 

```python
<button class="btn {{ button_class }}"
```

**Comment by hmpf** [comment]
I consider it a matter of building skills and habits so that you already can do the small change step by merged step when you later really need it, like changing something that is already in production without any downtime.

**Comment by hmpf** [comment]
In order to move the branch I need to open a new PR and close this one.

**Comment by hmpf** [comment]
Replaced by #139

File: src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html (hunk 2 above)
Comment (hmpf) [reply src/argus_htmx/templates/htmx/incidents/_incidents_update_menu.html:3]: Solved with new commit on #139.

**Pull request closed without merging**